### PR TITLE
feat: add dir into deps

### PR DESCRIPTION
**Pull request by beyondkmp**

### CODE REVIEW

**Review by mmaietta — APPROVED** [review]

File: packages/app-builder-lib/src/util/appFileCopier.ts [hunk 1]
```diff
@@ -191,25 +208,38 @@ function validateFileSet(fileSet: ResolvedFileSet): ResolvedFileSet {
...
-      source,
-      info.deps.map(it => it.name),
-      nodeModuleExcludedExts
-    )
+    const files = await copier.collectNodeModules(info, nodeModuleExcludedExts)
     result[index++] = validateFileSet({ src: source, destination, files, metadata: copier.metadata })
+
+    if (info.conflictDependency) {
+      for (const dep of info.conflictDependency) {
+        const source = platformPackager.info.appDir + path.sep + "node_modules" + path.sep + info.name + path.sep + "node_modules" + path.sep + dep.name
```
Comment (mmaietta) [inline packages/app-builder-lib/src/util/appFileCopier.ts:231]: Curious, how do we know that it's a double-nested `node_modules`? I'm worried about the hard-coded approach here, but wanted to understand the context here.

File: packages/app-builder-lib/src/util/packageDependencies.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { executeAppBuilderAsJson } from "./appBuilder"
 
 export function createLazyProductionDeps(projectDir: string, excludedDependencies: Array<string> | null) {
   return new Lazy(async () => {
-    const args = ["node-dep-tree", "--dir", projectDir]
+    const args = ["node-dep-tree", "--flatten", "--dir", projectDir]
```
Comment (mmaietta) [inline packages/app-builder-lib/src/util/packageDependencies.ts:6]: Is this a new arg `--flatten` you've implemented in app-builder or something that already exists? Curious as to what changes you made to app-builder for this to work? 🙂

File: packages/app-builder-lib/src/util/appFileCopier.ts (hunk 1 above)
Comment (beyondkmp) [reply packages/app-builder-lib/src/util/appFileCopier.ts:231]: All dependencies are placed directly into a map. If it's found that there's already a dependency with the same name but a different version in the map, it indicates a conflict. Currently, I'm placing the conflicting version directly into the node_modules of the current package.
https://github.com/beyondkmp/app-builder/blob/c85c25c911b6f5fc508afbbbd705b37220571b12/pkg/node-modules/nodeModuleCollector.go#L91-L107

The name conflictDependency might be somewhat misleading; we could consider renaming it.

File: packages/app-builder-lib/src/util/packageDependencies.ts (hunk 2 above)
Comment (mmaietta) [reply packages/app-builder-lib/src/util/packageDependencies.ts:6]: Great idea! Love it

**Pull request closed without merging**